cpp/ql/src/semmle/code/cpp/controlflow/internal/ConstantExprs.qll

```diff
@@ -136,6 +136,25 @@ private predicate impossibleDefaultSwitchEdge(Block switchBlock, DefaultCase dc)
                        val <= switchCaseRangeEnd(sc))))
 }
 
+/**
+ * Holds if the function `f` never returns due to not containing a return
+ * statement (explicit or compiler generated).  Reachability of return
+ * statements is not checked.
+ */
+private predicate nonReturningFunction(Function f)
```

Comment:
Does this have a different purpose than `potentiallyReturningFunction`?

Reply:
Not really.  `potentiallyReturningFunction` is likely to be more accurate because it uses `reachable`, which I deliberately avoided here as it would be recursive and performance is very sensitive here (it's actually non-monotonic recursion anyway).  We might be able to copy some useful implementation details from `potentiallyReturningFunction` however...

Reply:
I've updated the comment to explain this better, and made a few small improvements inspired by `potentiallyReturningFunction`.